numpy/doc/basics.py

```diff
@@ -260,6 +260,45 @@
 value is inside an array or not.  NumPy scalars also have many of the same
 methods arrays do.
 
+Overflow Errors
+===============
+
+The fixed size of NumPy numeric types may cause overflow errors when a value
+requires more memory than available in the data type. For example,
+:ref:`np.power` evaluates ``100 * 10 ** 8`` correctly for 64-bit integers,
+but gives 1874919424 (incorrect) for a 32-bit integer.
+
+    >>> np.power(100, 8, dtype=np.int64)
+    10000000000000000
+    >>> np.power(100, 8, dtype=np.int32)
+    1874919424
+
+The behaviour of NumPy and Python integer types differs significantly for
+integer overflows and may confuse users expecting NumPy integers to behave
+similar to Python's ``int``. Unlike NumPy, the size of Python's ``int`` is
+flexible. This means Python integers may expand to accomodate any integer and
+will not overflow.
+
+NumPy provides :ref:`numpy.iinfo` and :ref:`numpy.finfo` to verify the
+minimum or maximum values of NumPy integer and floating point values
+respectively ::
```

Comment:
the `numpy.power`, `numpy.iinfo` and `numpy.finfo` should **not** have `:ref:`, they should just be `single-back-tick content single-back-tick`.  I am not sure which domain is the correct one to get the links to appear, but `:ref:` apparently is not it. I tried this by doing `(cd ..; pip install .); make clean; make html` in the `doc` directory, and then `firefox build/html/user/basics.types.html` to view the result. The `pip install` is needed since the changes are part of the docstring picked up by `autodoc`, so the docstring (and therefore numpy) must be installed in the python used to build the docs

Reply:
> I am not sure which domain is the correct one to get the links to appear, but :ref: apparently is not it

The correct domain is `:obj:`

Reply:
I've removed the ``:ref:``'s entirely. That should make it simpler to read anyway. I've also rebased the commits to keep the history neat.